backend/stream.go

```diff
@@ -2,30 +2,53 @@ package backend
 
 import (
 	"context"
+	"encoding/json"
 )
 
 // StreamHandler handles streams.
 // This is EXPERIMENTAL and is a subject to change till Grafana 8.
 type StreamHandler interface {
-	// Called when a user tries to connect to a plugin/datasource managed channel.
-	CanSubscribeToStream(context.Context, *SubscribeToStreamRequest) (*SubscribeToStreamResponse, error)
-	// RunStream will be initiated by Grafana to consume a stream from a plugin.
-	// For streams with keepalive set this will only be called once the first client
-	// successfully subscribed to a stream channel. And when there are no longer any
-	// subscribers, the call will be terminated by Grafana.
+	// SubscribeStream called when a user tries to subscribe to a plugin/datasource
+	// managed channel path.
+	SubscribeStream(context.Context, *SubscribeStreamRequest) (*SubscribeStreamResponse, error)
```

Comment:
maybe "SubscribeToStream"  and "PublishToStream"?

Reply:
or just "Subscribe" and "Publish" ;)

Reply:
The namespace exists when looking at a single service, but for a real implementation, the functions are all implemented by the same struct -- this is why everything is named {Verb}{Service}.

Lets keep it as SubscribeStream -- so it matches the QueryData pattern